apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.html.ts

```diff
@@ -0,0 +1,557 @@
+        groupA,one@example.com<br />
+        groupA,two@example.com<br />
+        groupB,three@example.com<br />
+        groupB,four@example.com</code
+      >
+      <!-- Closing code tag not on its own line to improve copy/paste formatting -->
+      </div>
+      <div class="modal-body">
+        <p>
+          The <code>groupname</code> column should be a unique identifier for each group. To
+          change the grouping, link uids to the groupname.
+        </p>
+          <div class="form-group">
+            <div class="custom-file">
+              <input
+                type="file"
+                accept=".csv"
+                name="file"
+                class="custom-file-input"
+                id="uploadAssessmentGroupsFileInput"
+              />
+              <label class="custom-file-label" for="uploadAssessmentGroupsFileInput">
+                Choose CSV file
+              </label>
+            </div>
+          </div>
+          <div class="d-flex justify-content-end">
+            <div class="form-group mb-0">
+              <input type="hidden" name="__action" value="upload_assessment_groups" />
+              <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+              <button type="button" class="btn btn-secondary" data-dismiss="modal">
+                Cancel
+              </button>
+              <button type="submit" class="btn btn-primary">Upload</button>
+            </div>
+          </div>
+      </div>
+    `,
+  });
+}
+
+function AutoAssessmentGroupsModal({
+  groupMin,
+  groupMax,
+  csrfToken,
+}: {
+  groupMin: number;
+  groupMax: number;
+  csrfToken: string;
+}) {
+  return Modal({
+    id: 'autoAssessmentGroupsModal',
+    title: 'Auto new group setting',
+    formEncType: 'multipart/form-data',
+    body: html`
+      <div class="form-group">
+        <label for="formMin">Min number of members in a group</label>
+        <input
+          type="text"
+          value="${groupMin}"
+          class="form-control"
+          id="formMin"
+          name="min_group_size"
+        />
+      </div>
+      <div class="form-group">
+        <label for="formMax">Max number of members in a group</label>
+        <input
+          type="text"
+          value="${groupMax}"
+          class="form-control"
+          id="formMax"
+          name="max_group_size"
+        />
+      </div>
+    `,
+    footer: html`
+      <div class="form-group mb-0">
+        <input type="hidden" name="__action" value="auto_assessment_groups" />
+        <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+        <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+        <button type="submit" class="btn btn-primary">Group</button>
+      </div>
+    `,
+  });
+}
+
+function AddGroupModal({ csrfToken }: { csrfToken: string }) {
+  return Modal({
+    id: 'addGroupModal',
+    title: 'Add a group',
+    body: html`
+      <div class="modal-body">
```

Comment:
It shouldn't be necessary to include your own `modal-body` here since `Modal` already does that for you:

https://github.com/PrairieLearn/PrairieLearn/blob/be93dca96a59a1123ffc59dd0a8efc03a10661d7/apps/prairielearn/src/components/Modal.html.ts#L41

Reply:
Removed in [49b24ad](https://github.com/PrairieLearn/PrairieLearn/pull/9550/commits/49b24ad27a42ff43993f2210f2ac2a3d611c3f11).